utils/dev-tool/src/utils.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+use std::env::VarError;
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::{env, fs};
+
+#[macro_export]
+macro_rules! panic_if_unsuccessful {
+    ($success_code:expr) => {
+        if !$success_code.success() {
+            panic!()
+        }
+    };
+}
+
+pub fn tmp_dir() -> PathBuf {
+    PathBuf::from("/tmp")
+}
+
+pub fn android_dir<P: AsRef<Path>>(root: P) -> PathBuf {
+    root.as_ref().join("clients/android")
+}
+
+pub fn jni_lib_dir<P: AsRef<Path>>(root: P) -> PathBuf {
+    android_dir(root).join("core/src/main/jniLibs")
+}
+
+pub fn swift_dir<P: AsRef<Path>>(root: P) -> PathBuf {
+    root.as_ref().join("clients/apple/SwiftLockbookCore")
+}
+
+pub fn swift_inc<P: AsRef<Path>>(root: P) -> PathBuf {
+    swift_dir(root).join("Sources/CLockbookCore/include")
+}
+
+pub fn swift_lib<P: AsRef<Path>>(root: P) -> PathBuf {
+    swift_dir(root).join("Sources/CLockbookCore/lib")
+}
+
+pub fn core_dir<P: AsRef<Path>>(root: P) -> PathBuf {
+    root.as_ref().join("core")
+}
+
+pub fn local_env_path<P: AsRef<Path>>(root: P) -> PathBuf {
+    root.as_ref().join("containers/local.env")
+}
+
+pub fn test_env_path<P: AsRef<Path>>(root: P) -> PathBuf {
+    root.as_ref().join("containers/test.env")
+}
+
+pub fn get_commit_hash() -> String {
+    let commit_hash = Command::new("git")
```

Comment:
github can hand you this info, not sure if it's better or not:
1. in your yml you can like export github specific yaml values as env vars, you could get things like PR number for instance this way